nodejs/awsx/utils.ts

```diff
@@ -108,6 +108,26 @@ export function hasTrueBooleanMember(obj: any, memberName: string | number | sym
     return val === true;
 }
 
+/** @internal */
+export function getRegionFromOpts(opts: pulumi.CustomResourceOptions): pulumi.Output<aws.Region> {
+    if (opts.parent) {
+        return getRegion(opts.parent);
+    }
+
+    return getRegionFromProvider(opts.provider);
+}
+
+
+/** @internal */
+export function getRegion(res: pulumi.Resource): pulumi.Output<aws.Region> {
+    return getRegionFromProvider(res.getProvider("aws::"));
```

Comment:
Perhaps a comment explaining what this is doing? :-) . I think it is passing a "fake" type token that will convince `getProvider` to lookup the AWS provider?

Reply:
sure!